src/main/java/org/cdc/gov/sdp/aphl/AphlS3Endpoint.java

```diff
@@ -0,0 +1,145 @@
+package org.cdc.gov.sdp.aphl;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.aws.s3.S3Configuration;
+import org.apache.camel.component.aws.s3.S3Consumer;
+import org.apache.camel.component.aws.s3.S3Endpoint;
+import org.apache.camel.component.aws.s3.S3Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.S3ClientOptions;
+import com.amazonaws.services.s3.model.CreateBucketRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+
+/**
+ * The aws-s3 component is used for storing and retrieving objecct from Amazon
+ * S3 Storage Service.
+ */
+@UriEndpoint(producerOnly = true, scheme = "aphl-s3", title = "APHL Push only AWS S3 Storage Service", syntax = "aphl-s3:bucketNameOrArn", label = "cloud,file")
+public class AphlS3Endpoint extends DefaultEndpoint {
+
+	private static final Logger LOG = LoggerFactory.getLogger(S3Endpoint.class);
+
+	private AmazonS3 s3Client;
+
+	@UriPath(description = "Bucket name or ARN")
+	@Metadata(required = "true")
+	private String bucketNameOrArn; // to support component docs
```

Comment:
This is an s3 artifact, but I can refactor to separate them in the future if we decide to.